app/src/main/java/com/google/jetpackcamera/ui/CameraPermission.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetpackcamera.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+import com.google.jetpackcamera.R
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun CameraPermission(cameraPermissionState: PermissionState) {
+    Column {
+        val textToShow = if (cameraPermissionState.status.shouldShowRationale) {
+            stringResource(R.string.camera_permission_required_rationale)
+        } else {
+            stringResource(R.string.camera_not_available)
+        }
+
+        Text(textToShow)
+        Spacer(modifier = Modifier.height(8.dp))
+        Button(onClick = { cameraPermissionState.launchPermissionRequest() }) {
```

Comment:
Should the state object be performing actions? Not sure if that's the convention. Or should it be exposed as a param
`fun CameraPermission(cameraPermissionState: PermissionState, onClick: () -> Unit)`